common/namespace/registry.go

```diff
@@ -416,54 +422,91 @@ UpdateLoop:
 
 	// NOTE: READ REF BEFORE MODIFICATION
 	// ref: historyEngine.go registerNamespaceFailoverCallback function
-	r.callbackLock.Lock()
-	defer r.callbackLock.Unlock()
-	r.triggerNamespaceChangePrepareCallbackLocked()
-	r.cacheByID.Store(newCacheByID)
-	r.cacheNameToID.Store(newCacheNameToID)
-	r.triggerNamespaceChangeCallbackLocked(prevEntries, nextEntries)
-
-	// only update last refresh time when refresh succeeded
-	r.lastRefreshTime.Store(now)
+	r.makeVisibleWithCallbacks(func() (Namespaces, Namespaces) {
+		r.cacheLock.Lock()
+		defer r.cacheLock.Unlock()
+		r.cacheByID = newCacheByID
+		r.cacheNameToID = newCacheNameToID
+		return prevEntries, nextEntries
+	})
 	return nil
 }
 
-func (r *registry) updateNameToIDCache(
-	cacheNameToID cache.Cache,
-	name string,
-	id string,
-) {
-	cacheNameToID.Put(name, id)
+func (r *registry) updateNameToIDCache(c cache.Cache, name string, id string) {
+	c.Put(name, id)
 }
 
 func (r *registry) updateIDToNamespaceCache(
 	cacheByID cache.Cache,
 	id string,
-	record *Namespace,
+	newNS *Namespace,
 ) *Namespace {
-	prevCacheRec := cacheByID.Put(id, record)
-	if e, ok := prevCacheRec.(*Namespace); ok &&
-		record.notificationVersion > e.notificationVersion &&
+	prevCacheRec := cacheByID.Put(id, newNS)
+	if existingNS, ok := prevCacheRec.(*Namespace); ok &&
+		newNS.notificationVersion > existingNS.notificationVersion &&
 		r.globalNamespacesEnabled {
-		return e
+		return existingNS
```

Comment:
same as last one, use new/old to be consistent (would help reader to better understand)